Fire-Leds / Led sections

Liammurray19 opened the pull request:
Fire animation and switch case

Liammurray19 committed:
add file for the fire leds

I haven't tested it yet and I used chatgpt to translate it to java

Liammurray19 committed:
fire leds

Liammurray19 committed:
Tried to add FireLedMap

what I did was probable not correct I will work on it tomorrow

Liammurray19 committed:
Hopefully made it work

I will bring color alliance back after I test

Hsaunders603 committed:
Added testing programs

With consistent value increases for ease of documentation, 1by1, 2by2, 3by3, 4by4

Hsaunders603 committed:
Stopped using paths

Paths aren't useful, so made a switch to commands. Numbers need to be tested

Hsaunders603 committed:
Oops forgot to add it in

Hsaunders603 committed:
X and Y flipped

Lucimoot committed:
LED Sections (using If Statements)

first concept command, gonna try changing the code to use switch cases, so committing this in case i cause a dumpster fire

Lucimoot committed:
LED Sections (using Switch Cases)

does the same thing as the LED Sections with If statements but the code is slightly more efficient, I hope it works.

Lucimoot committed:
LED Sections (using Switch Cases)

Added break statements

msoucy commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -20,6 +22,12 @@ public class Led extends SmartSubsystemBase {
     // Length is expensive to set, so only set it once, then just update data
     AddressableLEDBuffer ledBuffer;
 
+    enum ledSection {
```
Capital L

msoucy commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -33,50 +41,80 @@ public Led(LedMap map) {
-            // Sets the specified LED to the RGB values for red
-            ledBuffer.setRGB(i, r, g, b);
+    public void setColor(int r, int g, int b, ledSection section) {
+        switch (section) {
```
Can you consolidate the duplicated code?

msoucy commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -33,50 +41,80 @@ public Led(LedMap map) {
+        return cmd("Make leds fire").onExecute(() -> {
+            for (int i = 0; i < FireLeds.heat.length; i++) {
+                Color color = new Color(FireLeds.heat[i] / 255.0, 0, 0);
+                ledBuffer.setLED(i, color);
```
Move this to outside the loop

msoucy commented on src/main/java/frc/robot/util/FireLeds.java:
```diff
@@ -0,0 +1,58 @@
+        // TODO: Add code to add delay
+    }
+
+    private static void setPixelHeatColor(int pixel, byte temperature) {
```
This logic should be done in the Led subsystem - or at least called from there. This function looks like it's supposed to convert a heat number to a color.

Lucimoot committed:
Auto stash before rebase of "main"

msoucy committed:
Undo auto related changes

Lucimoot committed:
Simplified duplicate code

changed the switch case into a if and else if statement and added one for loop that handles all operations rather than three separate for loops to handle each operation

Lucimoot committed:
Remove ChatGPT nonesense

begone ai code
